react-components/src/hooks/useModelIdRevisionIdFromModelOptions.test.tsx

```diff
@@ -64,53 +58,73 @@ describe(useModelIdRevisionIdFromModelOptions.name, () => {
   });
 
   test('returns modelId & revisionId for classic model option', async () => {
-    dependencies.useQueriedAddModelOptions.mockReturnValue([mockClassicModelResult]);
+    const resultMock = new Mock<UseQueryResult<Array<AddModelOptions<ClassicDataSourceType>>>>()
+      .setup((p) => p.data)
+      .returns([classicModelOption])
+      .setup((p) => p.isFetching)
+      .returns(false)
+      .setup((p) => p.isLoading)
+      .returns(false)
+      .setup((p) => p.isError)
+      .returns(false)
+      .setup((p) => p.isRefetching)
+      .returns(false)
+      .object();
+    dependencies.useQueriedAddModelOptions.mockReturnValue(resultMock);
     const { result } = renderHook(
       () => useModelIdRevisionIdFromModelOptions([classicModelOption]),
       { wrapper }
     );
 
     await waitFor(() => {
       expect(result.current.length).toBeGreaterThan(0);
-      expect(result.current[0].data).toBeDefined();
+      expect(result.current[0]).toBeDefined();
     });
 
-    assert(result.current[0]?.data !== undefined);
+    assert(result.current[0] !== undefined);
 
-    expect(isClassicIdentifier(result.current[0].data)).toBe(true);
-    expect(result.current[0]?.data).toMatchObject(classicModelOption);
+    expect(isClassicIdentifier(result.current[0])).toBe(true);
+    expect(result.current[0]).toMatchObject(classicModelOption);
   });
 
   test('returns modelId & revisionId for dm model option', async () => {
-    dependencies.useQueriedAddModelOptions.mockReturnValue([mockDMModelResult]);
+    const resultMock = new Mock<UseQueryResult<Array<AddModelOptions<ClassicDataSourceType>>>>()
+      .setup((p) => p.data)
+      .returns([{ modelId: 987, revisionId: 654 }])
+      .setup((p) => p.isFetching)
+      .returns(false)
+      .setup((p) => p.isLoading)
+      .returns(false)
+      .setup((p) => p.isError)
+      .returns(false)
+      .setup((p) => p.isRefetching)
+      .returns(false)
+      .object();
+    dependencies.useQueriedAddModelOptions.mockReturnValue(resultMock);
     const { result } = renderHook(() => useModelIdRevisionIdFromModelOptions([dmModelOption]), {
       wrapper
     });
 
     await waitFor(() => {
       expect(result.current.length).toBeGreaterThan(0);
-      expect(result.current[0].data).toBeDefined();
+      expect(result.current[0]).toBeDefined();
     });
 
     expect(isDM3DModelIdentifier(dmModelOption)).toBe(true);
-    expect(result.current[0]?.data).toMatchObject({
+    expect(result.current[0]).toMatchObject({
       modelId: 987,
       revisionId: 654
     });
   });
 
   test('returns empty array if any query is still fetching', () => {
-    const mockDMModelResult = new Mock<UseQueryResult<AddModelOptions<ClassicDataSourceType>>>()
+    const resultMock = new Mock<UseQueryResult<Array<AddModelOptions<ClassicDataSourceType>>>>()
       .setup((p) => p.data)
-      .returns({
-        modelId: 987,
-        revisionId: 654
-      })
+      .returns([])
```

Comment:
I don't really understand why we return an empty array here - don't we want to still use the classic model ID? Same for a few more cases in this file.